src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -312,6 +315,8 @@ public virtual bool TryInsert(TKey key, TValue value, out TValue? removed)
                     {
                         if (!RemoveLast(out removed) || _cacheMap.Count > _capacity)
                         {
+                            if (_cacheMap.Count > _capacity && _logger.IsWarn)
+                                _logger.Warn($"(TryInsert) Failed to remove the last item from the pool, the current state is {Count}/{_capacity}. {GetInfoAboutWorstValues}");
```

Comment:
Not necessarily failed - it is possible that `RemoveLast` was successful, but `Count` was higher than `_capacity` by more than 1 before this operation. But it is just a log so feel free to ignore